lib/cadet/accounts/accounts.ex

```diff
@@ -64,6 +65,21 @@ defmodule Cadet.Accounts do
     Repo.get(User, id)
   end
 
+  @doc """
+  Get student with given name and nusnet id or create one
+  """
+  def get_or_create_user(name, role, nusnet_id) do
```

Comment:
`nusnet_id` should be a required field since it is the only way users can login. Do make the corresponding change to `user_factory.ex`. Also remove the `validate_inclusion(:role, Role.__valid_values__())` validation since `cast` already performs that check.

After that, checking by `where(nusnet_id: ^nusnet_id)` should be sufficient since it's unique.

Reply:
Bump @sreycodes